src/helpers/glucose-and-meals/meals.ts

```diff
@@ -26,25 +43,71 @@ export function saveMeals(date: Date, meals: Meal[]): Promise<void> {
 }
 
 export function prepareMealForEditing(meal: Meal): Promise<void> {
-    return MyDataHelps.persistDeviceData([{ type: 'MealToEdit', value: JSON.stringify(meal) }]);
+    const mealReference: MealReference = { date: startOfDay(meal.timestamp), id: meal.id };
+    return MyDataHelps.persistDeviceData([{ type: 'MealToEdit', value: JSON.stringify(mealReference) }]);
 }
 
-export async function getMealToEdit(): Promise<Meal | undefined> {
-    let response = await MyDataHelps.queryDeviceData({
+export async function getMealToEdit(): Promise<MealReference | undefined> {
+    const response = await MyDataHelps.queryDeviceData({
         namespace: 'Project',
         type: 'MealToEdit'
     });
     if (response.deviceDataPoints.length > 0) {
-        return toMeal(JSON.parse(response.deviceDataPoints[0].value) as SerializedMeal);
+        return toMealReference(JSON.parse(response.deviceDataPoints[0].value) as SerializedMealReference);
     }
     return undefined;
 }
 
+export async function uploadMealImageFile(meal: Meal, file: File): Promise<void> {
+    const imageFileExtension = file.name.split('.').pop()!;
+
+    await MyDataHelps.uploadFile(file, 'MealImage', `${meal.id}.${imageFileExtension}`);
+
```

Comment:
Thanks for the suggestion.  Rather than default the file extension, I chose to just reject the promise.  I think this is a slim edge case and I'm comfortable with just rejecting those files.